File: 73주차/1913.py
n = int(input())
target = int(input())
dx = [-1,0,1,0]
dy= [0,1,0,-1]
K = n//2
g = [[0]*n for _ in range(n)]
x,y = K,K

cnt = 0
g[x][y] = 1
num = 2
d = 0
tx,ty= x,y # 찾을려고하는 숫자의 위치

while 0 <= x < n and 0 <= y < n: # 상 우 하(2번) 좌(2번) | 상(3번) 우(3번) 하(4번) 좌(4번),,
    if d % 2 == 0:
        cnt += 1
    for _ in range(cnt):
        x += dx[d]
        y += dy[d]
        if 0 <= x < n and 0 <= y < n:
            g[x][y] = num
            num += 1
        else:
            break
    d = (d+1)%4

for x in range(n):
    for y in range(n):
        print(g[x][y],end=' ')
        if g[x][y] == target:
            tx,ty = x+1,y+1
    print()

print(tx,ty)
# 달팽이 문제

File: 76주차/1913.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  ouput: process.stdout,
});
const inRange = (x, y) => {
  return 0 <= x && x < n && 0 <= y && y < n;
};
const solution = () => {
  const dx = [-1, 0, 1, 0];
  const dy = [0, 1, 0, -1];
  const arr = Array(n)
    .fill()
    .map((e) => Array(n).fill(0));
  let cnt = 0;
  let d = 0;
  let [x, y] = [parseInt(n / 2), parseInt(n / 2)];
  let k = 2;
  arr[x][y] = 1;
  let [ax, ay] = [0, 0];
  while (inRange(x, y)) {
    if (d % 2 === 0) {
      cnt += 1;
    }
    for (let i = 0; i < cnt; i++) {
      x += dx[d];
      y += dy[d];
      if (inRange(x, y)) {
        arr[x][y] = k;
        k += 1;
      } else break;
    }
    d = (d + 1) % 4;
  }
  for (let i = 0; i < n; i++) {
    const tmp = arr[i].join(" ");
    for (let j = 0; j < n; j++) {
      if (num === arr[i][j]) [ax, ay] = [i + 1, j + 1];
    }
    console.log(tmp);
  }
  console.log(ax, ay);
};
let n = null;
let num = null;
rl.on("line", (line) => {
  if (!n) {
    n = +line;
  } else {
    num = +line;
    rl.close();
  }
});
rl.on("close", () => {
  solution();
  process.exit();
});
//백준 달팽이
